src/retuner.cc

```diff
@@ -316,36 +316,61 @@ Retuner::process (int nfram, float const* inp, float* out)
 			// the circular input buffer limits it must be at
 			// least one fragment size.
 			dr = _cycle * (int)(ceilf (_frsize / _cycle));
+
```

Comment:
Prior to this PR the playback ratio was only taken in consideration for deciding _when_ to jump (`ph = ph / _frsize + 2 * _ratio - 10;`) and the jump size didn't depend on it (`dr = _cycle * (int)(ceilf (_frsize / _cycle));`). This lead to issues when the ratio was too high because the jumps could not compensate as fast as the playhead drifted (both with and without fast mode enabled), and with low ratios (in fast mode only) because jumps would occur too frequently.
As far as I could test this works better than the original (and seems easier to understand as well).